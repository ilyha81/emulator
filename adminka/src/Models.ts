
import {EDocumentStatus, EOperationStatus} from "./Enums";

/**
 * Интерфейс документа отправляемого при создании операции.
 */
export interface IDocument {
    id: string;
    docType: string;
}

/**
 * Интерфейс для Документа в операции подписания.
 * @prop {EDocumentStatus} status Статус документа.
 * @prop {string} errorCode Код ошибки если она произошла.
 * @prop {string} errorMsg Сообщение об ошибке, если она произошла.
 */
export interface IDocumentServer extends IDocument{
    status: EDocumentStatus;
    errorCode?: string;
    errorMsg?: string;
}

/**
 * Model of Operation on server
 */
export interface IOperationServer {
    id: string;
    status: EOperationStatus,
    timer: number,
    cryptoprofileId: string;
    documents: IDocumentServer[],
    createDate: string,
    dataSourceMode: string
}