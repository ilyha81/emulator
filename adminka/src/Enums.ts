/**
 * Documents LifeCycle Statuses.
 * @prop SENDING Отправлен на подписание.
 * @prop POSTED ХЗ
 * @prop DAY_LIMIT_ERROR ХЗ
 *
 */
export enum EDocumentStatus {
    SENDING = "SENDING",
    POSTED = "POSTED",
    DAY_LIMIT_ERROR = "DAY_LIMIT_ERROR"
}

/**
 * Operation LifeCycle Statuses.
 */
export enum EOperationStatus {
    POSTED_SUCCESS = "POSTED_SUCCESS",
    ANSWER_REFUSED = "ANSWER_REFUSED",
    PROCESSED_SUCCESS = "PROCESSED_SUCCESS",
    CONFIRMED = "CONFIRMED",
    PROCESSED_FAILED = "PROCESSED_FAILED",
    ANSWER_FAILED = "ANSWER_FAILED",
    SENDING = "SENDING",
    POSTED_FAILED = "POSTED_FAILED",
    SENDING_FAILED = "SENDING_FAILED",
}