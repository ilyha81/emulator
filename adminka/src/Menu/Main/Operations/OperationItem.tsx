import * as React from 'react';
import {IOperationServer} from "../../../Models";
import {IListTableItem, ListTable} from "../../../Components/ListTable/ListTable";

/**
 * OperationItem props.
 * @prop {IOperationServer} operation Model of server Operation.
 */
interface IOperationItem {
    operation: IOperationServer
}

/**
 * Drawing the element with Operation Data.
 */
export const OperationItem:React.FunctionComponent<IOperationItem> = ({operation}) => {

    const leftBlock: IListTableItem[] = [{name: 'ID:', value: operation.id},
        {name: 'Status:', value: operation.status},
        {name: 'TimeLeft:', value: operation.timer.toString()}];

    const rightBlock: IListTableItem[] = [{name: 'CreateDate:', value: operation.createDate},
        {name: 'Cryptoprofile:', value: operation.cryptoprofileId},
        {name: 'DocumentsCount:', value: operation.documents.length.toString()}];

    return <div className={'container-fluid border border-radius border-dark p-1 mb-1'} style={{backgroundColor: 'rgba(255,255,255,0.5)', overflow: 'hidden'}}>
        <div className={'row'}>
            <div className={'col-5'}>
                <ListTable items={leftBlock}/>
            </div>
            <div className={'col-7'}>
                <ListTable items={rightBlock}/>
            </div>
        </div>
    </div>
};