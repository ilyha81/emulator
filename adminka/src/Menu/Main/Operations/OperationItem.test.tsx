import * as React from 'react';
import {cleanup, render, screen} from '@testing-library/react';
import {OperationItem} from "./OperationItem";
import {IOperationServer} from "../../../Models";
import {EDocumentStatus, EOperationStatus} from "../../../Enums";

/**
 * Test data.
 */
const testOperation: IOperationServer = {
    id: 'testId',
    createDate: '2021-03-01, 17:00:01',
    cryptoprofileId: 'testCryptoprofile',
    dataSourceMode: 'testDataSourceMode',
    documents: [{id: 'docId', docType: 'docType', status: EDocumentStatus.POSTED}],
    status: EOperationStatus.POSTED_SUCCESS,
    timer: 100
};

beforeEach(() => {
    cleanup();
});


describe('Testing <OperationItem />', () => {
    test('Testing render', () => {
        const {container} = render(<OperationItem operation={testOperation}/>);
        expect(container).toMatchSnapshot();
    });

    test('Testing left block of data render', () => {
        render(<OperationItem operation={testOperation}/>);
        expect(screen.getByText('ID:'));
        expect(screen.getByText(testOperation.id));
        expect(screen.getByText('Status:'));
        expect(screen.getByText(testOperation.status));
        expect(screen.getByText('TimeLeft:'));
        expect(screen.getByText(testOperation.timer));
    });

    test('Testing right block of data render', () => {
        render(<OperationItem operation={testOperation}/>);
        expect(screen.getByText('CreateDate:'));
        expect(screen.getByText(testOperation.createDate));
        expect(screen.getByText('Cryptoprofile:'));
        expect(screen.getByText(testOperation.cryptoprofileId));
        expect(screen.getByText('DocumentsCount:'));
        expect(screen.getByText(testOperation.documents.length));
    })

});