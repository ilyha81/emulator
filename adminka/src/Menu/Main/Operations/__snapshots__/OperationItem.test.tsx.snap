// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Testing <OperationItem /> Testing render 1`] = `
<div>
  <div
    class="container-fluid border border-radius border-dark p-1 mb-1"
    style="background-color: rgba(255, 255, 255, 0.5); overflow: hidden;"
  >
    <div
      class="row"
    >
      <div
        class="col-5"
      >
        <table
          class="table table-striped table-hover table-dark m-0"
        >
          <tbody>
            <tr>
              <td>
                ID:
              </td>
              <td>
                testId
              </td>
            </tr>
            <tr>
              <td>
                Status:
              </td>
              <td>
                POSTED_SUCCESS
              </td>
            </tr>
            <tr>
              <td>
                TimeLeft:
              </td>
              <td>
                100
              </td>
            </tr>
          </tbody>
        </table>
      </div>
      <div
        class="col-7"
      >
        <table
          class="table table-striped table-hover table-dark m-0"
        >
          <tbody>
            <tr>
              <td>
                CreateDate:
              </td>
              <td>
                2021-03-01, 17:00:01
              </td>
            </tr>
            <tr>
              <td>
                Cryptoprofile:
              </td>
              <td>
                testCryptoprofile
              </td>
            </tr>
            <tr>
              <td>
                DocumentsCount:
              </td>
              <td>
                1
              </td>
            </tr>
          </tbody>
        </table>
      </div>
    </div>
  </div>
</div>
`;
